rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3827,6 +3829,14 @@ static void rmw_fini_cs(CddsCS * cs)
   dds_delete(cs->sub->rdcondh);
   dds_delete(cs->sub->enth);
   dds_delete(cs->pub->enth);
+  if (cs->sub) {
+    delete cs->sub;
+    cs->sub = NULL;
+  }
+  if (cs->pub) {
+    delete cs->pub;
+    cs->pub = NULL;
+  }
```

Comment:
Just above it already dereference `cs->sub`, so we know that it is not a null pointer (same for `cs->pub`) and after checking the call sites, I do believe that it only calls `rmw_fini_cs` with a fully initialised `CddsCS`. That said, I think it is a good idea to check. But then is should be:

```suggestion
  if (cs->sub) {
    dds_delete(cs->sub->rdcondh);
    dds_delete(cs->sub->enth);
    delete cs->sub;
    cs->sub = NULL;
  }
  if (cs->pub) {
    dds_delete(cs->pub->enth);
    delete cs->pub;
    cs->pub = NULL;
  }
```

Reply:
Thanks for your suggestions. I believe you're right.
Actually, I also prefer using `unique_ptr`. I'll update it that makes us clearly.